File: packages/dart/lib/src/word/rand_space.dart
import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// What is beyond the sky, from the moon to a galaxy.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.space] to `randWordDetails` — Dart has
/// no overloads, and a second function per theme would be a second wall of them.
///
/// ```dart
/// randSpace(language: WordLanguage.ko, count: 3); // [은하, 혜성, 북극성]
/// randSpace(language: WordLanguage.en, count: 3); // [Galaxy, Comet, Nebula]
/// ```
List<String> randSpace({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,
}) => randWord(
  language: language,
  theme: WordTheme.space,
  count: count,
  realism: realism,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
);

File: packages/javascript/lib/word/randSpace.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * What is beyond the sky, from the moon to a galaxy.
 *
 * `randWord({ theme: 'space' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randSpace({ language: 'ko', count: 3 }); // ['은하', '혜성', '북극성']
 * randSpace({ language: 'en', count: 3 }); // ['Galaxy', 'Comet', 'Nebula']
 */
export function randSpace(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Space words with the language and theme each one came from.
 *
 * @example
 * randSpace({ language: 'ko', output: 'detail' });
 * // [{ word: '은하', language: 'ko', theme: 'space' }]
 */
export function randSpace(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randSpace(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('space', options);
}
